ps_emailsubscription.php

```diff
@@ -1353,25 +1353,27 @@ public function hookDisplayAdminCustomersForm()
 
     public function hookActionDeleteGDPRCustomer($customer)
     {
-        if (!empty($customer['email']) && Validate::isEmail($customer['email'])) {
-            $sql = 'DELETE FROM ' . _DB_PREFIX_ . "emailsubscription WHERE email = '" . pSQL($customer['email']) . "'";
-            if (Db::getInstance()->execute($sql)) {
-                return json_encode(true);
+        if (isset($customer['email'])) {
+            if (!empty($customer['email']) && Validate::isEmail($customer['email'])) {
+                $sql = "DELETE FROM "._DB_PREFIX_."emailsubscription WHERE email = '".pSQL($customer['email'])."'";
+                if (Db::getInstance()->execute($sql)) {
+                    return json_encode(true);
+                }
+                return json_encode($this->l('Newsletter subscription: Unable to delete customer using email.'));
             }
-
-            return json_encode($this->trans('Newsletter subscription: no email to delete, this customer has not registered.', array(), 'Modules.Emailsubscription.Admin'));
         }
     }
 
     public function hookActionExportGDPRData($customer)
     {
-        if (!Tools::isEmpty($customer['email']) && Validate::isEmail($customer['email'])) {
-            $sql = 'SELECT * FROM ' . _DB_PREFIX_ . "emailsubscription WHERE email = '" . pSQL($customer['email']) . "'";
-            if ($res = Db::getInstance()->ExecuteS($sql)) {
-                return json_encode($res);
+        if (isset($customer['email'])) {
+            if (!Tools::isEmpty($customer['email']) && Validate::isEmail($customer['email'])) {
```

Comment:
```suggestion
            if (!empty($customer['email']) && Validate::isEmail($customer['email'])) {
```

Reply:
I implemented your suggestion, thank you for your help :)